### PR TITLE
fix: update mirror-demo to leverage json api

### LINKED ISSUE #1261: Update mirror-demo to use JSON toolbox definitions instead of xml
**Category**
 - Examples

**Component**

mirror-demo

**Is your feature request related to a problem? Please describe.**

Many of the examples and plugins use XML toolboxes instead of the new JSON API. We should have our example code use the new APIs to best model what other developers can do with Blockly.

**Describe the solution you'd like**

Update the code in blockly-samples to use the JSON toolbox API.

**Additional context**

To work on this issue, first read the [background information](https://developers.google.com/blockly/guides/contribute/samples) about how to contribute to samples and verify your changes.

1. Learn about [defining a toolbox](https://developers.google.com/blockly/guides/configure/web/toolbox) and read about the XML format vs the JSON format.
2. In blockly-samples, navigate to the mirror-demo example's directory. Some examples can be started with `npm run start` and some can be opened directly by opening the `index.html` file in a browser. The README may give specific instructions, or you can look in the `package.json` to see what scripts are defined. Note you may need to run `npm install` in the example directory. Either way, open the example's demo page to observe the toolbox and its contents.
3. Write a new copy of the toolbox definition in the JSON format instead of the XML format. Where exactly it should be declared will depend on the example. It will most likely make sense to declare the toolbox definition close to the location where it is used, i.e. where `Blockly.inject` is called.
4. Refresh the demo page and make sure the toolbox looks the same as it did before. You can compare against the version in master on the [samples site](https://google.github.io/blockly-samples/) to make sure the behavior hasn't changed. Note that not all demos are published on the samples site so it may not be there.
5. Follow the rest of the instructions in the Contributing docs to verify your change and open a PR.

### PR DESCRIPTION
**Pull request by melanie-t**
Fixes #1261

Changes made
- Update `mirror-demo` Blockly dependency to version 8
- Replace xml `mirror-demo` example to leverage new JSON API

### CODE REVIEW

File: examples/mirror-demo/index.html [hunk 1]
```diff
@@ -34,6 +34,7 @@
     </tr>
   </table>
 
+  <!-- XML toolbox implementation
```
Comment (alschmiedt) [inline examples/mirror-demo/index.html]: Do you mind removing this old toolbox definition?

**Comment by alschmiedt** [comment]
Hello! Once you remove the toolbox definition we should be able to get this merged in : ) Thank you for your work on this!

File: examples/mirror-demo/index.html [hunk 2]
```diff
@@ -47,13 +48,77 @@
...
+          "fields": {
+            "VAR": {
+              "name": "k"
+            }
+          }
+        },
+      ]
+    };
+  
+    // // Inject primary workspace.
```
Comment (maribethb) [inline examples/mirror-demo/index.html]: nit: could you remove the extra comment slashes?

File: examples/mirror-demo/index.html [hunk 3]
```diff
@@ -47,13 +48,77 @@
     <block type="variables_get"><field name="VAR">i</field></block>
     <block type="variables_get"><field name="VAR">j</field></block>
     <block type="variables_get"><field name="VAR">k</field></block>
-  </xml>
-
+  </xml> -->
```
Comment (maribethb) [inline examples/mirror-demo/index.html]: You're good to go ahead and remove this code instead of commenting it out! It isn't needed anymore :)

**Review by maribethb — COMMENTED** [review]
This looks great! Just some minor changes requested and then this will be good to go! Thank you for the PR!

**Comment by maribethb** [comment]
whoops sorry didn't see Abby's review before I sent mine off :)

File: examples/mirror-demo/index.html (hunk 1 above)
Comment (melanie-t) [reply examples/mirror-demo/index.html]: Done, thank you for the review!

File: examples/mirror-demo/index.html (hunk 2 above)
Comment (melanie-t) [reply examples/mirror-demo/index.html]: Done, thank you for catching this!

**Review by alschmiedt — APPROVED** [review]
Looks good to me(LGTM), thank you for your work on this! 🎉

**Comment by rachel-fenichel** [comment]
Thanks @melanie-t for contributing to Blockly!

**Pull request merged**